File: packages/backend/src/question/question.mapper.ts
import { QuestionDto } from 'src/question/dto/question.dto';
import { QuestionModel } from 'src/question/model/question.model';

export class QuestionMapper {
  static toDto(model: QuestionModel): QuestionDto {
    return {
      id: model.id,
      createdAt: model.createdAt,
      question: model.question,
      questionType: model.questionType,
      answers: model.answers ?? null,
      correctAnswer: model.correctAnswer,
    };
  }

  static toDtoList(models: QuestionModel[]): QuestionDto[] {
    return models.map(this.toDto);
  }
}

File: packages/backend/src/main/kotlin/quizzem/backend/features/question/mapper/QuestionMapper.kt
package quizzem.backend.features.question.mapper

import quizzem.backend.features.answer.mapper.toDto
import quizzem.backend.features.question.api.dto.QuestionDto
import quizzem.backend.features.question.model.QuestionModel


fun QuestionModel.toDto(): QuestionDto {
    return QuestionDto(
        id = requireNotNull(id),
        createdAt = createdAt,
        updatedAt = updatedAt,
        text = text,
        type = type,
        answers = answers.map { it.toDto() }
    )
}